File: python/potega.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# a0 = 1
# a1 = a
# an = a * ... * a (n-czynników) dla N+ - {0, 1}


def potega_it(podstawa, wykladnik):
    wynik = 1
    for i in range(wykladnik):
        wynik = wynik * podstawa
    return wynik

# a0 = 1 - warunek brzegowy
# an = a(n- 1) * a dla n > 0


def potega_rek(podstawa, wykladnik):
    if wykladnik == 0:
        return 1
    return potega_rek(podstawa, wykladnik - 1) * podstawa


def main(args):
    podstawa = int(input('Podaj podstawę: '))
    wykladnik = int(input('Podaj wykładnik: '))

    assert type(podstawa) == int
    assert type(wykladnik) == int

    assert potega_it(100, 0) == 1
    assert potega_it(100, 1) == 100
    assert potega_it(2, 3) == 8
    assert potega_rek(100, 0) == 1
    assert potega_rek(100, 1) == 100
    assert potega_rek(2, 3) == 8

    print('Wynik: ', potega_it(podstawa, wykladnik))
    print('Wynik: ', potega_rek(podstawa, wykladnik))
    return 0


if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: cpp/potega.cpp
/*
 * potega.cpp
 */


#include <iostream>

using namespace std;

int potega_it(int podstawa, int wykladnik)
{
    int wynik = 1;
    int i = 1;
    while (i <= wykladnik)
    {
        wynik = wynik * podstawa;
        i = i + 1;
    }
    return wynik;
}   

int potega_rek(int podstawa, int wykladnik)
{
    if (wykladnik == 0)
    {
        return 1;
    }
    return potega_rek(podstawa, wykladnik - 1) * podstawa;
}

int main(int argc, char **argv)
{
    int podstawa = 0;
    int wykladnik = 0;
    
    cout << "Podaj podstawę: " ;
    cin >> podstawa ; 
    cout << "Podaj wykładnik: " ;
    cin >> wykladnik ; 
    
    cout << "Wynik: " << potega_it(podstawa, wykladnik) << endl;
    cout << "Wynik: " << potega_rek(podstawa, wykladnik) << endl;
    
    return 0;
}


